src/extension/features/general/privacy-mode/index.js

```diff
@@ -52,6 +52,12 @@ export class PrivacyMode extends Feature {
     this.updatePrivacyMode();
   }
 
+  observe() {
+    if (!this.shouldInvoke) return;
```

Comment:
few issues here:

1. this isn't actually calling `shouldInvoke`, it's just checking for the existence of the function (both result in _always true_)
2. rather than calling `shouldInvoke`, we should check against the `changedNodes` set to see if there's anything that's changed that warrants our feature to run again (the sidebar)
3. it might be easier to fix this feature just by adding a `onRouteChanged` hook instead of `observe` since it sounds like that's when this actually becomes a problem?

Reply:
yes, sorry, still trying to learn how this all works.  I will try that hook instead, i also didn't realize the shouldInvoke call wasn't necessary in this case to get it to work.  I was trying to figure out how to even tap into those, observe fixed the problem for me so 🤷🏼‍♂️  I'll fix this and send it in for review again.

Reply:
I made the change you requested and tested that it is working and only calling when the url path changes.

<img width="252" alt="Screen Shot 2020-08-13 at 10 21 58 PM" src="https://user-images.githubusercontent.com/26221344/90210509-bf80df80-ddb3-11ea-92a0-89ecf7f0925b.png">